geomesa-tools/src/main/scala/org/locationtech/geomesa/tools/ingest/ScaldingConverterIngestJob.scala

```diff
@@ -0,0 +1,119 @@
+    Map(
+      dsp.zookeepersParam.getName -> args(IngestParams.ZOOKEEPERS),
+      dsp.instanceIdParam.getName -> args(IngestParams.ACCUMULO_INSTANCE),
+      dsp.tableNameParam.getName  -> args(IngestParams.CATALOG_TABLE),
+      dsp.userParam.getName       -> args(IngestParams.ACCUMULO_USER),
+      dsp.passwordParam.getName   -> args(IngestParams.ACCUMULO_PASSWORD),
+      dsp.authsParam.getName      -> args.optional(IngestParams.AUTHORIZATIONS),
+      dsp.visibilityParam.getName -> args.optional(IngestParams.VISIBILITIES),
+      dsp.mockParam.getName       -> args.optional(IngestParams.ACCUMULO_MOCK)
+    ).collect{ case (key, Some(value)) => (key, value); case (key, value: String) => (key, value) }
+
+  lazy val sft = {
+    val ret = SimpleFeatureTypes.createType(featureName, sftSpec)
+    args.optional(IngestParams.INDEX_SCHEMA_FMT).foreach { indexSchema =>
+      ret.getUserData.put(Constants.SFT_INDEX_SCHEMA, indexSchema)
+    }
+    ret
+  }
+
+  // non-serializable resources.
+  class Resources {
+    val ds = DataStoreFinder.getDataStore(dsConfig).asInstanceOf[AccumuloDataStore]
+    lazy val fw = ds.getFeatureWriterAppend(featureName, Transaction.AUTO_COMMIT)
+    val converter = SimpleFeatureConverters.build[String](sft, ConfigFactory.parseString(converterConfig))
+    val callback = converter.processWithCallback(counter = counter)
+    def release(): Unit = {
+      logger.trace("Releasing ingest resources")
+      converter.close()
+      fw.close()
+    }
+  }
+
+  def printStatInfo() {
+    Mode.getMode(args) match {
+      case Some(Local(_)) =>
+        logger.info(getStatInfo(counter.getSuccess, counter.getFailure, "Local ingest completed, total lines:"))
+      case Some(Hdfs(_, _)) =>
+        logger.info("Ingest completed in HDFS mode")
+      case _ =>
+        logger.warn("Could not determine job mode")
+    }
+  }
+
+  def getStatInfo(successes: Int, failures: Int, pref: String): String = {
+    val successPvsS = if (successes == 1) "feature" else "features"
+    val failurePvsS = if (failures == 1) "feature" else "features"
+    val failureString = if (failures == 0) "with no failures" else s"and failed to ingest: $failures $failurePvsS"
+    s"$pref ${counter.getLineCount}, ingested: $successes $successPvsS, $failureString."
+  }
+
+  // Check to see if this an actual ingest job or just a test.
+  if (!isTestRun) {
+    new MultipleUsefulTextLineFiles(pathList: _*).using(new Resources)
+      .foreach('line) { (cres: Resources, s: String) =>
+        processLine(cres, s)
+      }
+  }
+
+  private def processLine(resources: Resources, s: String) =
+    resources.callback(s)
+      .foreach { sf =>
+        val toWrite = resources.fw.next()
+        toWrite.setAttributes(sf.getAttributes)
+        toWrite.getIdentifier.asInstanceOf[FeatureIdImpl].setID(sf.getID)
+        toWrite.getUserData.putAll(sf.getUserData)
+        resources.fw.write()
+      }
+
+  def runTestIngest(lines: Iterator[String]) = {
```

Comment:
I think I can probably manage that...I added a test that actually executes the code. So I'll get ride of this thing.
https://github.com/locationtech/geomesa/pull/719/files#diff-ac19bd149f34d32e2a89db3822c0078fR146